### PR TITLE
Hw1

### PR DESCRIPTION
**Pull request by 6ECCMEPTHIoIIJ**

### CODE REVIEW

File: src/main/java/edu/hw1/Task1.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.regex.Pattern;
+
+public final class Task1 {
+    private final static String SEPARATOR = ":";
+    private final static int SECONDS_PER_MINUTE = 60;
+
+    private final static String TIME_STRING_REGEX = "\\d{2,}:[0-5][0-9]";
+    private final static Pattern TIME_PATTERN = Pattern.compile(TIME_STRING_REGEX);
+
+    private Task1() {
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task1.java:11]: Удалите не используемые методы

File: src/main/java/edu/hw1/Task1.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public static long minutesToSeconds(String timeString)
+        throws NullPointerException {
+        if (timeString == null) {
+            throw new NullPointerException();
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task1.java]: Более подходит другое исключение

File: src/test/java/edu/hw1/Task2Test.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package edu.hw1;
+
+import java.util.stream.Stream;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class Task2Test {
```
Comment (vasilmik) [inline src/test/java/edu/hw1/Task2Test.java:10]: Проверьте  corner case и увидите что алгоритм не всегда работает верно

File: src/main/java/edu/hw1/Task6.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    private final static int FOUR_DIGIT_NUMBER_DIGITS_COUNT = 4;
+
+    private final static int DECIMAL_BASE = 10;
+
+    private Task6() {
+
+    }
+
+    private static boolean hasNumberAllDifferentDigits(int num) {
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task6.java:17]: Нельзя руками заносить все значения, а если их будет 100 тыс, необходимо переписать через алгоритм

File: src/test/java/edu/hw1/Task8Test.java [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+            NullPointerException.class,
+            () -> Task8.knightBoardCapture(null)
+        );
+
+        assertNotNull(throwable);
+    }
+
+    static @NotNull Stream<DeskTestArgs> nonCapturedDeskProviderFactory() {
+        return Stream.of(
+            new DeskTestArgs(new boolean[][] {
```
Comment (vasilmik) [inline src/test/java/edu/hw1/Task8Test.java:43]: Сделайте нормальный тест как в задаче

File: src/main/java/edu/hw1/Task8.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        return false;
+    }
+
+    public static boolean knightBoardCapture(boolean[][] desk)
+        throws NullPointerException {
+        if (desk == null) {
+            throw new NullPointerException();
+        }
+
+        return !isKnightOverlapped(desk, -KNIGHT_LONG_MOVE_DISTANCE, KNIGHT_SHORT_MOVE_DISTANCE)
```
Comment (vasilmik) [inline src/main/java/edu/hw1/Task8.java:32]: Проверьте эти варианты

File: src/main/java/edu/hw1/Task1.java (hunk 1 above)
Comment (6ECCMEPTHIoIIJ) [reply src/main/java/edu/hw1/Task1.java:11]: private Task1() - приватный конструктор утилитного класса. Если его удалить, то код не пройдет checkstyle.

File: src/main/java/edu/hw1/Task6.java (hunk 4 above)
Comment (6ECCMEPTHIoIIJ) [reply src/main/java/edu/hw1/Task6.java:17]: Исправил. 
Изначально был написан алгоритм с циклом, но чуть позже пришел к выводу, что для 4-хзначного числа перебрать 9 вариантов вручную не такая проблема (потому что их немного), при этому такой вариант намного эффективнее с точки зрения времени исполнения.

Comment (6ECCMEPTHIoIIJ) [reply src/main/java/edu/hw1/Task6.java:17]: однако, если бы вариантов было больше, такая идея в голову бы не пришла

File: src/test/java/edu/hw1/Task2Test.java (hunk 3 above)
Comment (6ECCMEPTHIoIIJ) [reply src/test/java/edu/hw1/Task2Test.java:10]: Действительно, тест упал, когда проверил на Integer.MIN_VALUE из-за переполнения типа при взятии модуля. По итогу возвращалось некорректное значение 1. 
Сделал отрицательные числа не валидными

File: src/test/java/edu/hw1/Task8Test.java (hunk 5 above)
Comment (6ECCMEPTHIoIIJ) [reply src/test/java/edu/hw1/Task8Test.java:43]: Добавил тесты из примера к заданию. 
Не совсем понял, что не так с моими тестами (кроме того, что true false невозможно читать). Там проверяются варианты на пустое поле и поле, где все кони не могут друг на друга напасть, а также все варианты на взаимное расположение коней (их всего 4). На каждый из 4 вариантов проверяются 5 вариантов их расположения относительно доски: в центре доски, в одном из 4-х углов

File: src/main/java/edu/hw1/Task8.java (hunk 6 above)
Comment (6ECCMEPTHIoIIJ) [reply src/main/java/edu/hw1/Task8.java:32]: Не могли бы вы пояснить, что именно мне необходимо проверить?

Comment (6ECCMEPTHIoIIJ) [reply src/main/java/edu/hw1/Task8.java:32]: В комментарии к Task8Test расписал, какие именно варианты я проверяю
